tests/h/streamer/tweens_test.py

```diff
@@ -0,0 +1,43 @@
+from unittest import mock
+
+import pytest
+
+from h.streamer import close_db_session_tween_factory
+
+
+class TestCloseTheDBSessionTweenFactory:
+    def test_it_calls_the_handler(
+        self, close_db_session_tween, handler, pyramid_request
+    ):
+        close_db_session_tween(pyramid_request)
+
+        handler.assert_called_once_with(pyramid_request)
+
+    def test_it_closes_the_db_session(self, close_db_session_tween, pyramid_request):
+        close_db_session_tween(pyramid_request)
+
+        pyramid_request.db.close.assert_called_once_with()
+
+    def test_it_closes_the_db_session_if_an_exception_is_raised(
+        self, close_db_session_tween, handler, pyramid_request
+    ):
+        handler.side_effect = RuntimeError("test_error")
+
+        with pytest.raises(RuntimeError, match="^test_error$"):
+            close_db_session_tween(pyramid_request)
+
+        pyramid_request.db.close.assert_called_once_with()
+
+    @pytest.fixture
+    def close_db_session_tween(self, handler):
+        return close_db_session_tween_factory(handler, mock.sentinel.registry)
+
+    @pytest.fixture
+    def handler(self):
+        handler = mock.create_autospec(lambda request: None)  # pragma: nocover
+        return handler
+
+    @pytest.fixture
+    def pyramid_request(self, pyramid_request):
+        pyramid_request.db = mock.MagicMock(spec_set=["close"])
```

Comment:
I'd normally use `create_autospec()` instead of `spec_set`. `spec_set` has the disadvantage that the tests contain hard-coded knowledge about the interface of the thing being mocked (in this case: that it has a `close()` method) and that needs to be kept in sync with the real `request.db`.

In this case I don't think sqlalchemy's `session.close()` is likely to ever change, and while running `create_autospec` on sqlalchemy's session class would probably work it might also be slow.

Reply:
If we're worried about autospec being slow we should probably time it. My intuition is it's totally dominated by database access, and not worth worrying about. Either way we should probably know for sure